### ISSUE TITLE
[SFTTraining] Eval cause stuck

### ISSUE THREAD

**Issue by ListentoMe0112** [huggingface/open-r1#361]
The processes get stuck in the evaluation process during training. I'm really new to LLMs, so I don't really know what happened. Any help would be appreciated!

This is my device and status while stucking

![Image](https://github.com/user-attachments/assets/ae9f2cfb-546b-48a3-a508-74f4ff246af8)
py-spy info for two process
![Image](https://github.com/user-attachments/assets/a60d3628-352f-4b01-88d4-c24d45242efa)
![Image](https://github.com/user-attachments/assets/98bd6bf3-46f3-4f13-a845-9333b639ed04)

## accelerate_config
compute_environment: LOCAL_MACHINE
debug: false
deepspeed_config:
  deepspeed_multinode_launcher: standard
  offload_optimizer_device: none
  offload_param_device: none
  zero3_init_flag: true
  zero3_save_16bit_model: true
  zero_stage: 3
distributed_type: DEEPSPEED
downcast_bf16: 'no'
machine_rank: 0
main_training_function: main
mixed_precision: fp16
num_machines: 1
num_processes: 2
rdzv_backend: static
same_network: true
tpu_env: []
tpu_use_cluster: false
tpu_use_sudo: false
use_cpu: false

## config_demo
model_name_or_path: Qwen/Qwen2.5-1.5B-Instruct
model_revision: main
torch_dtype: float16

dataset_name: HuggingFaceH4/Bespoke-Stratos-17k
dataset_configs:
- all
preprocessing_num_workers: 2

bf16: true
do_eval: true
eval_strategy: steps
eval_steps: 1
gradient_accumulation_steps: 1
gradient_checkpointing: true
gradient_checkpointing_kwargs:
  use_reentrant: false
hub_model_id: Qwen2.5-1.5B-Open-R1-Distill
hub_strategy: every_save
learning_rate: 2.0e-05
log_level: info
logging_steps: 5
logging_strategy: steps
lr_scheduler_type: cosine
packing: true
max_seq_length: 2048
max_steps: -1
num_train_epochs: 1
output_dir: data/Qwen2.5-1.5B-Open-R1-Distill
overwrite_output_dir: true
per_device_eval_batch_size: 1
per_device_train_batch_size: 1
push_to_hub: true
report_to:
- wandb
save_strategy: "no"
seed: 42
warmup_ratio: 0.1

The eval_steps set to be one for quick reproduce.

## start command
nohup accelerate launch --config_file recipes/accelerate_configs/zero3.yaml src/open_r1/sft.py --config recipes/Qwen2.5-1.5B-Instruct/sft/config_demo.yaml > train.log 2>&1 

## nvcc info
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2024 NVIDIA Corporation
Built on Thu_Mar_28_02:18:24_PDT_2024
Cuda compilation tools, release 12.4, V12.4.131
Build cuda_12.4.r12.4/compiler.34097967_0

**Comment by hjxwhy** [comment]
you can update trl to latest, i found trl has fix the large tensor gather cause the stuck

**Comment by ListentoMe0112** [comment]
> you can update trl to latest, i found trl has fix the large tensor gather cause the stuck

how can I do this?

**Comment by ListentoMe0112** [comment]
> you can update trl to latest, i found trl has fix the large tensor gather cause the stuck

What should I do? Will setting eval_steps to a large number work?

**Comment by ListentoMe0112** [comment]
> you can update trl to latest, i found trl has fix the large tensor gather cause the stuck

Is trl related to grpo? I'm still in the sft stage. Where is the trl-related code?
